option.go

```diff
@@ -240,6 +230,26 @@ func WithSendMaxBytes(max int) Option {
 	return &sendMaxBytesOption{Max: max}
 }
 
+// WithIdempotency declares the idempotency of the procedure. This can determine
+// whether a procedure call can safely be retried, and may affect which request
+// modalities are allowed for a given procedure call.
+//
+// In most cases, you should not need to manually set this. It is normally set
+// by the code generator for your schema.
+func WithIdempotency(idempotencyLevel IdempotencyLevel) Option {
+	return &idempotencyOption{idempotencyLevel: idempotencyLevel}
+}
+
+// WithGetURLMaxBytes allows using HTTP Get requests for requests whose URLs
+// would fall under the provided maximum number of bytes. Please note that this
+// feature has some performance impact for requests made to RPCs which are
+// applicable, as a message may need to be marshalled twice.
+//
+// If this value is set to zero, no HTTP Get requests will be allowed.
```

Comment:
Should clarify that this only applies to idempotent requests with no side effects. It'd also be good if we clarified _why_ this is important (middle boxes limit URL size) and what a reasonable value might be (what are the default limits for NGINX, Apache, Envoy, etc?).

Maybe `WithCacheableRequests(maxURLSize int)` or `WithHTTPGet(maxURLSize int)` would be a better name?

Reply:
I chose `WithHTTPGet`, although I'm not really certain what the best approach is. 596851957872611bdcf249a139c456302a1ac374